model/testing/src/icon4py/model/testing/pytest_hooks.py

```diff
@@ -123,12 +124,45 @@ def pytest_runtest_setup(item: pytest.Item) -> None:
     """Apply test item filters as the final test setup step."""
 
     item_marker_filters = filters.item_marker_filters
-    for marker_name in set(m.name for m in item.own_markers) & item_marker_filters.keys():
+    for marker_name in set(m.name for m in item.iter_markers()) & item_marker_filters.keys():
         item_filter = item_marker_filters[marker_name]
         if item_filter.condition(item):
             item_filter.action()
 
 
+_name_from_fullname_pattern = re.compile(
+    r"""
+        ::(?P<class>[A-Za-z_]\w*)       # capture class name
+        (?::: [A-Za-z_]\w*              # skip method name
+        (?:\[(?P<params>[^\]]+)\])? )   # optional parameterization
+        """,
+    re.VERBOSE,
+)
+
+
+def _name_from_fullname(fullname: str) -> str:
+    match = _name_from_fullname_pattern.search(fullname)
+    if match is None:
+        return fullname  # assume already fixed
+    class_name = match.group("class")
+    params = match.group("params")
+    return f"{class_name}[{params}]" if params else class_name
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_benchmark_group_stats(config, benchmarks, group_by):
+    """
+    Replace 'name' of pytest benchmarks with a shorter name for better readability in bencher.
+
+    See also the 'pytest_benchmark_update_json' hook below.
+    """
+    result = collections.defaultdict(list)
+    for bench in benchmarks:
+        bench["name"] = _name_from_fullname(bench["fullname"])
+        result[bench["name"]].append(bench)
+    yield result.items()
```

Comment:
As discussed offline, renaming the tests should be moved again to the `StencilTest` `__init_subclass__` method.